virttest/qemu_vm.py

```diff
@@ -104,14 +104,20 @@ def qemu_proc_term_handler(vm, monitor_exit_status, exit_status):
     Callback function to detect QEMU process non-zero exit status and
     push VMExitStatusError to background error bus.
 
-    :param vm: VM name.
+    :param vm: VM object.
     :param monitor_exit_status: True to push VMUnexpectedExitError instance
         with calltrace to global error event bus.
     :param exit_status: QEMU process exit status.
     """
+    for _ in vm.devices.temporary_image_snapshots[::-1]:
```

Comment:
How about using `set` to represent `temporary_image_snapshots`, then code would be:
```python
for snapshot in vm.devices.temporary_image_snapshots:
    try:
        os.unlink(snapshot)
    except:
        pass
vm.devices.temporary_image_snapshots.clear()
```
and it just requires to change the method from `append` to `add` when adding items.

Reply:
ok, will update it , BTW there is no attribute` clear()` in python2,
would use `del vm.devices.temporary_image_snapshots`

Reply:
@YongxueHong `list` doesn't have the `clear` method but `set` do, please see https://docs.python.org/2.7/library/stdtypes.html?highlight=set#frozenset.clear

Reply:
ok , will update it
Thanks.